modules/gdscript/gdscript_functions.cpp

```diff
@@ -722,29 +722,18 @@ void GDScriptFunctions::call(Function p_func, const Variant **p_args, int p_arg_
 		} break;
 		case STR_TO_VAR: {
 			VALIDATE_ARG_COUNT(1);
-			if (p_args[0]->get_type() != Variant::STRING) {
-				r_error.error = Variant::CallError::CALL_ERROR_INVALID_ARGUMENT;
-				r_error.argument = 0;
-				r_error.expected = Variant::STRING;
-				r_ret = Variant();
-				return;
-			}
```

Comment:
I think you can keep the input type check. Not passing a string to `str2var` is fishy anyway.

Reply:
Passing non-`String` is an error indeed. Is it grave enough to crash, though? With error message in console you can at least easily locate the bug.

Reply:
> Is it grave enough to crash, though? With error message in console you can at least easily locate the bug.

A crash is much easily located because the editor will stop and show you the line where it falls. The error in console can be easily missed. Note that this does not really crash the game on release mode.